src/libraries/Microsoft.Extensions.Configuration/tests/FunctionalTests/ConfigurationTests.cs

```diff
@@ -987,6 +987,41 @@ void ReloadLoop()
             }
         }
 
+        [Fact]
+        public async Task ReloadingFileFiresOnChangeEvent()
```

Comment:
This test should be a `Theory` and you should put `./reload.json` and `reload.json` as `InlineData`.